.github/copilot-instructions.md

```diff
@@ -0,0 +1,171 @@
+  - **net10.0 branch**: Use the latest .NET 10 SDK
+  - **etc.**: Each feature branch correlates to its respective .NET version
+- **C#** and **XAML** for application development
+- **Cake build system** for compilation and packaging
+- **MSBuild** with custom build tasks
+- **xUnit** for testing
+
+## Development Environment Setup
+
+### Prerequisites
+
+#### Linux Development (Current Environment)
+For .NET installation on Linux, follow the official Microsoft documentation:
+* https://learn.microsoft.com/en-us/dotnet/core/install/linux
+
+#### Additional Requirements
+- **OpenJDK 17** for Android development
+- **VS Code** with .NET MAUI Dev Kit extension
+- **Android SDK** for Android development
+
+### Initial Repository Setup
+
+1. **Clone and navigate to repository:**
+   ```bash
+   git clone https://github.com/dotnet/maui.git
+   cd maui
+   ```
+
+2. **Restore tools and build tasks (REQUIRED before opening IDE):**
+   ```bash
+   dotnet tool restore
+   dotnet build ./Microsoft.Maui.BuildTasks.slnf
+   ```
+
+## Project Structure
+
+### Important Directories
+- `src/Core/` - Core MAUI framework code
+- `src/Controls/` - UI controls and components
+- `src/Essentials/` - Platform APIs and essentials
+- `src/TestUtils/` - Testing utilities and infrastructure
+- `docs/` - Development documentation
+- `eng/` - Build engineering and tooling
+- `.github/` - GitHub workflows and configuration
+
+### Platform-Specific Code Organization
+- **Android** specific code is inside folders labeled `Android`
+- **iOS** specific code is inside folders labeled `iOS`
+- **MacCatalyst** specific code is inside folders named `MacCatalyst`
+- **Windows** specific code is inside folders named `Windows`
+
+### Platform-Specific File Extensions
+- Files with `.windows.cs` will only compile for the Windows TFM
+- Files with `.android.cs` will only compile for the Android TFM
+- Files with `.ios.cs` will only compile for the iOS and MacCatalyst TFM
+- Files with `MacCatalyst.cs` will only compile for the MacCatalyst TFM
+
+### Sample Projects
+```
+├── Controls 
+│   ├── samples
+│   │   ├── Maui.Controls.Sample
+│   │   ├── Maui.Controls.Sample.Sandbox
+├── Essentials 
+│   ├── samples
+│   │   ├── Essentials.Sample
+├── BlazorWebView 
+│   ├── samples
+│   │   ├── BlazorWinFormsApp
+│   │   ├── BlazorWpfApp
+```
+
+- `src/Controls/samples/Maui.Controls.Sample` - Full gallery sample with all controls and features
+- `src/Controls/samples/Maui.Controls.Sample.Sandbox` - Empty project for testing/reproduction
+- `src/Essentials/samples/Essentials.Sample` - Essentials API demonstrations (non-UI MAUI APIs)
+- `src/BlazorWebView/samples/` - BlazorWebView sample applications
+
+## Development Workflow
+
+### Building
+
+#### Using Cake (Recommended)
+```bash
+# Build everything
+dotnet cake
+
+# Pack NuGet packages
+dotnet cake --target=dotnet-pack
+```
+
+### Testing and Debugging
+
+#### Running Tests
+```bash
+# Run all tests
```

Comment:
Remove this instruction about running all tests
The section below has a list of all test projects you will be able to run